### PR TITLE
Fix errors regarding to Mautic form handling

### PR DESCRIPTION
**Pull request by Moongazer**
In my opinion the changes should be backward compatible to v10, but not sure about v9... Would v9 still be a requirement?

### CODE REVIEW

File: Classes/Domain/Finishers/MauticFinisher.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ public function __construct(string $finisherIdentifier = '')
     protected function executeInternal()
     {
         $formDefinition = $this->finisherContext->getFormRuntime()->getFormDefinition()->getRenderingOptions();
-        $mauticId = !empty($this->parseOption('mauticId')) ? (int)$this->parseOption('mauticId') : $formDefinition['mauticId'];
+        $mauticId = !empty($this->parseOption('mauticId')) ? (int)$this->parseOption('mauticId') : $formDefinition['mauticId'] ?? 0;
```
Comment (IchHabRecht) [inline Classes/Domain/Finishers/MauticFinisher.php]: I would consider to simplify this to 
```
$mauticId = (int)$this->parseOption('mauticId') ?: (int)($formDefinition['mauticId'] ?? 0);
```

File: Classes/Domain/Model/Dto/YamlConfiguration.php [hunk 2]
```diff
@@ -115,6 +115,7 @@ public function __construct()
 
     protected function getYamlConfiguration(): array
     {
+        if (!file_exists($this->fileName)) return [];
```
Comment (IchHabRecht) [inline Classes/Domain/Model/Dto/YamlConfiguration.php]: I coudn't find any reason for this check as everything seems to work as intended without. Do you mind to share the reason for this change?

File: Classes/Domain/Repository/FormRepository.php [hunk 3]
```diff
@@ -50,7 +50,7 @@ public function getForm(int $identifier): array
 
     public function getAllForms(): array
     {
-        return $this->formsApi->getList('', 0, 999)['forms'] ?: [];
+        return $this->formsApi->getList('', 0, 999)['forms'] ?? [];
```
Comment (IchHabRecht) [inline Classes/Domain/Repository/FormRepository.php]: This change introduces a new problem: when the 'forms' value is set but not an array, the function will throw an array.
My purpose:
```
return ($this->formsApi->getList('', 0, 999)['forms'] ?? []) ?: [];
```

File: Classes/Hooks/MauticFormHook.php [hunk 4]
```diff
@@ -65,10 +66,15 @@ class MauticFormHook implements LoggerAwareInterface
     public function __construct(FormPersistenceManagerInterface $formPersistenceManager = null)
     {
         if ($formPersistenceManager === null) {
-            $formPersistenceManager = GeneralUtility::makeInstance(FormPersistenceManager::class);
-            $formPersistenceManager->initializeObject();
-            $formPersistenceManager->injectResourceFactory(GeneralUtility::makeInstance(ResourceFactory::class));
-            $formPersistenceManager->injectYamlSource(GeneralUtility::makeInstance(YamlSource::class));
+            // @todo: remove if v10 is not supported anymore
```
Comment (IchHabRecht) [inline Classes/Hooks/MauticFormHook.php]: To be honest I don't see any reason here to try to resolve the dependency on your own. Just use dependency injection here.

**Comment by IchHabRecht** [comment]
Hi @Moongazer,

Thank you for your contribution to the Mautic TYPO3 extension. I allowed myself to attach some remarks and would kindly ask you to resolve the mentioned points.

According to the composer information, the extension doesn't need to be compatible to TYPO3 versions below 10.4. 

If you have any further questions, don't hesitate to contact me.

**Comment by Moongazer** [comment]
> I allowed myself to attach some remarks and would kindly ask you to resolve the mentioned points.

Hi, where did you attached them? I don't see any code-line comments or other points yet :)

**Comment by IchHabRecht** [comment]
> > I allowed myself to attach some remarks and would kindly ask you to resolve the mentioned points.
> 
> Hi, where did you attached them? I don't see any code-line comments or other points yet :)

Thank you very much for your feedback. I missed to click the correct button to publish my comments.

File: Classes/Domain/Model/Dto/YamlConfiguration.php (hunk 2 above)
Comment (Moongazer) [reply Classes/Domain/Model/Dto/YamlConfiguration.php]: As far as I remember, an exception was thrown here as the extension was installed the first time and not configured yet, so the YAML file didn't exist.

File: Classes/Domain/Repository/FormRepository.php (hunk 3 above)
Comment (Moongazer) [reply Classes/Domain/Repository/FormRepository.php]: True, thanks!

File: Classes/Hooks/MauticFormHook.php (hunk 4 above)
Comment (Moongazer) [reply Classes/Hooks/MauticFormHook.php]: From my understanding it uses DI already, but the manual resolving is for the case that `$formPersistenceManager` is null, which seems to happen after an update to v11:
```
Too few arguments to function TYPO3\CMS\Form\Mvc\Persistence\FormPersistenceManager::__construct(), 0 passed in /var/www/html/public/typo3/sysext/core/Classes/Utility/GeneralUtility.php on line 3220 and exactly 6 expected
```

File: Classes/Domain/Model/Dto/YamlConfiguration.php (hunk 2 above)
Comment (IchHabRecht) [reply Classes/Domain/Model/Dto/YamlConfiguration.php]: I tried to reproduce this but it the exception was caught correctly.

File: Classes/Hooks/MauticFormHook.php [hunk 5]
```diff
@@ -23,6 +23,7 @@
 use Bitmotion\Mautic\Transformation\FormField\Prototype\ListTransformationPrototype;
 use Psr\Log\LoggerAwareInterface;
 use Psr\Log\LoggerAwareTrait;
+use TYPO3\CMS\Core\Information\Typo3Version;
```
Comment (IchHabRecht) [inline Classes/Hooks/MauticFormHook.php]: Not necessary anymore

**Pull request merged**